### DISCUSSION TITLE
Splitting library and site concerns

### DISCUSSION THREAD

**Discussion by julealgon** [fluentassertions/fluentassertions#1804] [Ideas]
I was reading the [latest release notes](https://github.com/fluentassertions/fluentassertions/releases/tag/6.5.1) and noticed this there:
> Make the site work with Ruby 3 and update catchphrase and sponsors by @dennisdoomen in https://github.com/fluentassertions/fluentassertions/pull/1801

I'd like to say that, at least from a library consumer perspective, I'm only interested in library-related updates when reading the release notes. And I'm not saying that to feel entitled or anything: I'd assume most of the community would agree.

At the same time, I fully understand you guys wanting to keep track of everything that happens on the site side as well.

With that in mind, have you considered splitting site and library concerns into perhaps 2 different repositories, each with their own release schedules/versions/etc?

**Comment by dennisdoomen** [comment]
That's why we have consumer-facing release notes on the website. See https://fluentassertions.com/releases/

**Reply by julealgon** [reply]
While this is obviously up to you to decide, I think this is less than ideal. Github provides us with notifications about the release, so that's the go-to place (at least to me) to go to for the updates. The site has no such functionality. Also, github releases is "closer to the library" than the site is, which is also why people would naturally go there for checking the release notes. 

I'm not implying with this that there should be no release notes info in the site, but that the info on Github should be focused on the library as well. FluentAssertions is a library, and not a site. Additionally, it doesn't depend on the site to work. So one would expect to see library-related changes in the release notes.

The coupling between site and library is also undesirable if you want to make changes only to the site, for example. This would force a version bump/nuget/etc which would not make sense. I assume that, in this case, you just don't create a release at all and document the change once the library has an update? Again... obviously less than ideal from my perspective.

From your message, the feeling I get is that you are currently fine with how this is. I'd kindly ask you to reconsider, in that case.

**Reply by eNeRGy164** [reply]
As the release notes are automatically generated by GitHub, maybe it supports some filtering functionality? To skip a certain tag that is added to site/maintenance related issues?

**Reply by julealgon** [reply]
> As the release notes are automatically generated by GitHub

I feel like this should be an extra incentive to separate the site and library in different repositories.

**Reply by dennisdoomen** [reply]
In the past we had the site and the code separated. But the problem was that this caused the documentation to get out of sync with the code, especially with upcoming features that are on `develop` and are not ready to be released. In the end, having the code and the documentation in the same PR has been a great choice for us. That it means you sometimes get notifications like happened this time is less of an issue for me. But maybe we can remove those items after the release notes are generated.

**Reply by julealgon** [reply]
Noted. Personally, I think it would be nice to decouple "documentation contents" (should _definitely_ live with the library) from "documentation presentation" (the website, which IMHO should be separate). An example of that would be MSDN itself vs the libraries it documents: that only relies on XML docs from the code but the visuals that consume those are in a different place.

I assume you have already considered that though, and consciously opted for the simpler approach of having them together.

If there is no native mechanism to filter out parts of the release notes though, trying to achieve that would be hacky and I wouldn't recommend that to anyone even if it would simplify the notes. But again that's on you folks to decide.

**Comment by jnyrup** [comment]
I know it's in _no_ way intended, but comparing us to MSDN is a little unfair in terms of manpower/resources.

As Dennis mentions, to keep things easier for us, documentation/examples goes in with the PR changing the code.
As the website is build from `master` this had the side-effect that documentation could refer to unreleased features.
To improve on this we just recently started using a `develop` branch again, such that `master` more closely reflects the code available from nuget.org.

There is definitely room for improvement in the auto-generated GitHub release notes.
When releasing 6.5.1 I did consider whether to include the chore PRs, but ended up taking the lazier approach and not curating the release notes.
I think we do want to keep every PR listed to also acknowledge people contributing non-functional PRs.
I'm grateful for every typo of mine that anyone takes the time fix.

Seems we can [configure](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#configuration-options) the auto-generated GitHub release notes to group the changes by labels.
For inspiration see the looks of https://github.com/CommunityToolkit/dotnet/releases.

**Reply by julealgon** [reply]
> I know it's in no way intended, but comparing us to MSDN is a little unfair in terms of manpower/resources.

Yeah I mean... I made the comment just in regards to separation of documentation contents and presentation of such contents, but not on how much extra effort that would take. I actually didn't even know it would necessarily require more effort or if it would just be an organization thing, where you change a few things around once and you are done.

From what you say, it does seem to be a bit more involved, and it is understandable you would not want to devote resources to that.

> Seems we can [configure](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#configuration-options) the auto-generated GitHub release notes to group the changes by labels.
For inspiration see the looks of https://github.com/CommunityToolkit/dotnet/releases.

That certainly looks _really_ good from a consumer perspective. It has the exact things I would like to see and in a very nice order (basically respecting major/minor/patch).

> When releasing 6.5.1 I did consider whether to include the chore PRs, but ended up taking the lazier approach and not curating the release notes.
I think we do want to keep every PR listed to also acknowledge people contributing non-functional PRs.

I understand why you'd want to do that even though I personally wouldn't care much (for instance, if I did a refactoring/formatting change in the repo, I would personally not expect my name to be in the release notes document since those changes do not affect consumers, and consumers are the most important users of the release notes). But if possible, as per the example you provided, it would indeed be really nice if they were in some sort of lower priority "Other" section.
